### PR TITLE
⚡ Admin optimization & refactoring

### PR DESCRIPTION
**Pull request by ivan-kozhan**

### CODE REVIEW

File: api/Application.Core/Data/DataExtensions.cs [hunk 1]
```diff
@@ -62,7 +62,43 @@ public static IPageableList<TSource> ToPageableList<TSource>(this IOrderedQuerya
...
+        public static async Task<IPageableList<TSource>> ToPageableListAsync<TSource>(this IOrderedQueryable<TSource> source, int pageNumber = 1, int pageSize = 0)
+        {
+            var totalCount = await source.CountAsync();
+
+            var pageIndex = Math.Max(0, pageNumber - 1);
+
+			// Hangs if ToListAsync is used
+            var pageItems = await Task.Run(() =>
+            {
+                return source
```
Comment (hawkrai) [inline api/Application.Core/Data/DataExtensions.cs]: Can we use ToListAsync()? Please avoid using Task.Run

File: api/Application.Core/Data/DataExtensions.cs [hunk 2]
```diff
@@ -62,7 +62,43 @@ public static IPageableList<TSource> ToPageableList<TSource>(this IOrderedQuerya
 			};
 		}
 
-		public static void Add<TModel, TDataContext>(this TDataContext dataContext, TModel model)
+        public static async Task<IPageableList<TSource>> ToPageableListAsync<TSource>(this IOrderedQueryable<TSource> source, int pageNumber = 1, int pageSize = 0)
```
Comment (hawkrai) [inline api/Application.Core/Data/DataExtensions.cs:65]: Do we still use this extension? If yes. we should change it.

File: api/LMPlatform.Data/Repositories/LecturerRepository.cs [hunk 3]
```diff
@@ -33,6 +33,7 @@ public void DeleteLecturer(Lecturer lecturer)
                 if (deletedLecturer == null)
                     return;
                 deletedLecturer.IsActive = false;
+                deletedLecturer.DeletionDate = DateTime.UtcNow;
```
Comment (hawkrai) [inline api/LMPlatform.Data/Repositories/LecturerRepository.cs]: AddedOn, DeletedOn - more general naming approach

File: modules/admin/src/app/modules/adminPanel/lectors/lectors.component.ts [hunk 4]
```diff
@@ -205,15 +275,28 @@ export class LectorsComponent implements OnInit {
...
+
+  getStatusTooltip(lecturer: Professor) {
+    let tooltip = "";
+    
+    if (lecturer.IsActive) {
+      return;
+    }
+
+    if (lecturer.DeletionDate != null) {
+      return "Удален\r\n" +
```
Comment (hawkrai) [inline modules/admin/src/app/modules/adminPanel/lectors/lectors.component.ts:298]: Should we use translation here?

File: api/Application.Core/Data/DataExtensions.cs (hunk 1 above)
Comment (ivan-kozhan) [reply api/Application.Core/Data/DataExtensions.cs]: If we use ToListAsync() method freezes after this line is executed. Shall i use ToList() without Task.Run then?

File: modules/admin/src/app/modules/adminPanel/lectors/lectors.component.ts (hunk 4 above)
Comment (ivan-kozhan) [reply modules/admin/src/app/modules/adminPanel/lectors/lectors.component.ts:298]: There is a front-end developer who will be working on translation of admin module.

**Pull request merged**